### ISSUE TITLE
Test timeouts

### ISSUE THREAD

**Issue by lukechampine** [fortytw2/leaktest#5]
leaktest does not play well with `go test -timeout=5s`. Regardless, waiting 5 seconds for goroutines to shut down seems excessive. My expectation was that goroutines would be checked as soon as the deferred function was called.
I suggest adding an optional timeout argument to `Check`, or providing some other way of delaying the check. Note that if `Check` did not delay by default, you could implement this behavior yourself:

``` go
func TestFoo(t testing.T) {
    checkFn := leaktest.Check(t)
    defer func() {
        time.Sleep(5 * time.Second)
        checkFn()
    }()
}
```

Furthermore, this would give you more fine-grained control over when the check executes. For example, you could wait on a channel before checking. The only reason I propose adding an optional timeout argument is because it seems like a common case.

**Comment by fortytw2** [comment]
Ah, great suggestion :+1: 

I think I'll move the canonical imports to use `gopkg.in` shortly, mark the current version of the package as `.v1` and then make a `.v2` that removes the default timeout and adds a `CheckTimeout(*testing.T, time.Duration)` that works like you'd suggest. Sound OK?
